models/components.go

```diff
@@ -76,17 +67,17 @@ type Component struct {
 	Parents        []string                  `json:"parents,omitempty" gorm:"-"`
 }
 
-func (c *Component) GetStatus() ComponentStatus {
+func (c *Component) GetStatus() types.ComponentStatus {
```

Comment:
Though I would just use GetStatus() method from Summary

```go
func (c *Component) GetStatus() types.ComponentStatus {
	return c.Summary.GetStatus()
}
```
but they have different logics. Not sure if it's correct to do it this way.

Reply:
Here, I see that component's status is being set using Summary's `GetStatus` method

https://github.com/flanksource/canary-checker/blob/master/pkg/topology/run.go#L399

Reply:
Hmm, not sure about it. @moshloop Can you chime in here